src/api-umbrella/utils/interval_lock.lua

```diff
@@ -0,0 +1,78 @@
+local lock = require "resty.lock"
+
+local _M = {}
+
+--- Only one thread can execute fn through its execution duration
+-- A lock (mutex) is held while the function executes and released at the end
+-- @param name - a unique identifier for this lock (automatically namespaced)
+-- @param fn - function to execute if the lock can be held
+-- @return ok, error
+_M.mutex_exec = function(name, fn)
+  local check_lock = lock:new("locks", {["timeout"] = 0})
+  local _, lock_err = check_lock:lock("mutex_exec:" .. name)
+  if lock_err then
+    return true, nil
+  else
+    local results = pcall(fn)
```

Comment:
What are your thoughts about checking the status of the `pcall` and logging an error here if it didn't succeed? `pcall` is used to call functions and capture any errors they might raise. But since the errors don't bubble up further it means that unless you check the status of the `pcall` result and do something with it, errors will silently be swallowed. So anytime I use pcall, I usually check the result immediately afterwards and log if there was an error. Here's one example from the previous usage: https://github.com/NREL/api-umbrella/blob/3920e2852ae4ff074c205f606462050fd4310ea2/src/api-umbrella/proxy/jobs/distributed_rate_limit_puller.lua#L92-L95

Also, just FYI, the return signature on `pcall` is a bit funny. The first variable it returns will always be the status of whether the call succeeded. If the call failed, then the second variable will be the error message about what failed. If the call succeeded, then the second variable will be the original return value of the function you were calling (in our cases, I'm not sure we have a need for the real return value from these background jobs, since I'm not sure they return anything, but I suppose that could change at some point).